src/Resources/ResourceManager/Implementation/Deployments/GetAzureSubscriptionDeploymentCmdlet.cs

```diff
@@ -37,25 +38,25 @@ public class GetAzureDeploymentCmdlet : ResourceManagerCmdletBase
         internal const string DeploymentNameParameterSet = "GetByDeploymentName";
 
         [Alias("DeploymentName")]
-        [Parameter(Position = 0, ParameterSetName = GetAzureDeploymentCmdlet.DeploymentNameParameterSet, Mandatory = false,
+        [Parameter(Position = 0, ParameterSetName = GetAzureSubscriptionDeploymentCmdlet.DeploymentNameParameterSet, Mandatory = false,
             HelpMessage = "The name of deployment.")]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
         [Alias("DeploymentId", "ResourceId")]
-        [Parameter(ParameterSetName = GetAzureDeploymentCmdlet.DeploymentIdParameterSet, Mandatory = false,
+        [Parameter(ParameterSetName = GetAzureSubscriptionDeploymentCmdlet.DeploymentIdParameterSet, Mandatory = true,
```

Comment:
Changing an optional parameter to mandatory is a breaking change, and it cannot be checked in until next breaking change release which is ~May.
I agree that it should be mandatory, but my suggestion is to add a breaking change attribute and make the change later.

Reply:
This is also the cause to CI failure

Reply:
This shouldn't be breaking. "Id" is the only parameter in this parameter set. If you don't specify the "id" parameter, you'll land with the default parameter set which is "GetByDeploymentName".

Reply:
I can change it back to "mandatory" to false if that's the right thing..

Reply:
Right.. then it's a false positive. Please read [Debugging StaticAnalysis Errors](https://github.com/Azure/azure-powershell/blob/master/documentation/Debugging-StaticAnalysis-Errors.md#breaking-changes) to suppress the warning.